web-app/src/components/GridGallery.tsx

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import { GridGalleryItem } from './GridGalleryItem';
+
+import '../styles/grid-gallery.css';
+
+export enum PhotoSize {
+    Small,
+    Medium,
+    Large
+}
+
+export interface PhotoObject {
+    width: PhotoSize;
+    height: PhotoSize;
+    url: string;
+}
+
+export interface GridGalleryProps {
+    photos: string[];
+}
+
+export interface GridGalleryState {
+    photoObjects: PhotoObject[];
+}
+
+export class GridGallery extends React.Component<GridGalleryProps, GridGalleryState> {
```

Comment:
We should at somepoint convert this to a react functional component. I feel it might make it easier for us haha.